app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -522,7 +548,6 @@ def create_vm(args = { })
         clone_vm(args)
       else
         vm = new_vm(args)
```

Comment:
Then you can raise the error here (probably with some message):
```suggestion
        vm = new_vm(args)
        raise ArgumentError if errors.any?
```